extendeddatasquare.go

```diff
@@ -10,26 +10,21 @@ import (
 type ExtendedDataSquare struct {
 	*dataSquare
 	originalDataWidth uint
-	codec             CodecType
 }
```

Comment:
I didn't actually have to remove this codec as a field, but removing it made the api consistent. Every time a codec is needed, the user is required to provide one. This forces the user to consciously choose an already used codec, or a new instance of that codec.